devguide/howtos/add-function.md

```diff
@@ -63,8 +63,19 @@ The parameters passed to call are all optional (except ~name and ~in_product_sin
   permitted to merely be a single value rather than a list of values. If no
   ~result is specified, the default is unit.
 
+- The ~doc parameter describes what it is, the message is doing.
+
+- The ~hide_from_docs parameter prevents te message from being included in the document ation when generated. This is indicated by a bool.
+
+- The ~pool_internal parameter is used to indicate if the message should be callable by all or only internal hosts. This is indicated by a bool. 
+
+- The ~errs parameter is a list of possible exceptions that the message can raise.
+
 - The ~allowed_roles parameter is used for access control (see below).
 
+- The parameter ~lifecycle takes in an array of (Status, version, doc) to indicate the lifecycle of message type. This takes over from ~in_oss_since which indicated the release that the message type was introduced in.
```

Comment:
For new functions this should be left empty, as there is code will write the correct version it's introduced. You can see this by  running make after adding the functions, the changes will be visible in the file `ocaml/idl/datamodel_lifecycle.ml`

Reply:
So should I add a sentence to the end explaining that this parameter is added programatically?